framework/src/problems/EigenProblem.C

```diff
@@ -316,6 +316,26 @@ EigenProblem::scaleEigenvector(const Real scaling_factor)
   _nl_eigen->update();
 }
 
+void
+EigenProblem::initEigenvector(const Real initial_value)
+{
+  std::vector<VariableName> var_names = getVariableNames();
+  for (auto & vn : var_names)
+  {
+    MooseVariableFEBase & var = getVariable(0, vn);
+    if (var.parameters().get<bool>("eigen"))
+      for (unsigned int vc = 0; vc < var.count(); ++vc)
+      {
+        std::set<dof_id_type> var_indices;
+        _nl_eigen->system().local_dof_indices(var.number() + vc, var_indices);
+        for (const auto & dof : var_indices)
+          _nl_eigen->solution().set(dof, initial_value);
```

Comment:
I know this code is the same as the one in the old executioner, but I'd let you be aware that this is not very right. Setting all dofs to this initial value does not make the solution flat in cases of MONOMIAL. It is right for LAGRANGE, but not always right. I was sloppy during the old implementation because libMesh does not have a simple way for us to set dofs so that the solution is constant over the domain.

Reply:
I will check `MONOMIAL`